### PR TITLE
Removed ion-c dependency

### PR DESCRIPTION
**Pull request by zslayton**
This PR:
* Adds a native Rust implementation of the `ion dump` command.
* Removes the `ion-c` git submodule, which was required to allow
`ion dump` to wrap the ion-c CLI's format converter utility.
* Removes the custom build logic (`build.rs`) that was required to build
`ion-c`.
* Upgrades the CLI's dependency on `ion-rs` to v0.11.0,
eliminating a transitive dependency on `ion-c`.
* Upgrades the CLI's dependency on `ion-schema` to v0.3.0, eliminating a
transitive dependency on `ion-c`.

This means that users building the `ion-cli` no longer need cmake or
clang, and the overall build time has been drastically reduced.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: .gitmodules [hunk 1]
```diff
@@ -1,3 +0,0 @@
-[submodule "ion-c"]
-	path = ion-c
-	url = https://github.com/amzn/ion-c.git
```
Comment (zslayton) [inline .gitmodules:3]: 🗺️ We no longer need to pull in `ion-c` as a submodule; we now provide a native implementation of `ion dump`.

File: src/bin/ion/commands/beta/inspect.rs [hunk 2]
```diff
@@ -498,7 +505,7 @@ impl<'a> IonInspector<'a> {
         // If it is a container, `inspect_level` will handle stepping into it and writing any
         // nested values.
         if !self.reader.ion_type().unwrap().is_container() {
-            self.hex_buffer.push_str(" ");
+            self.hex_buffer.push(' ');
```
Comment (zslayton) [inline src/bin/ion/commands/beta/inspect.rs:508]: 🗺️ All other changes in this file (`inspect.rs`) are tweaks to satisfy `clippy`.

File: src/bin/ion/commands/dump.rs [hunk 3]
```diff
@@ -70,30 +34,134 @@ pub fn app() -> CommandConfig {
...
+        }
+        unrecognized => unreachable!(
+            "'format' was '{}' instead of 'pretty', 'text', or 'binary'",
+            unrecognized
+        ),
+    }
+}
+
+/// Writes each value encountered in the Reader to the provided Writer.
+fn write_all_values<W: Writer>(reader: &mut Reader, writer: &mut W) -> IonResult<()> {
```
Comment (zslayton) [inline src/bin/ion/commands/dump.rs:105]: 🗺️ TODO: This really should be a method on the `Writer` trait.

**Review by zslayton — COMMENTED** [review]
🗺️ PR tour

**Review by desaikd — APPROVED** [review]
🚀

**Review by jobarr-amzn — APPROVED** [review]

**Pull request merged**